pkg/operator/csi/controller/credentials/resource.go

```diff
@@ -0,0 +1,118 @@
+package credentials
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/client-go/config/clientset/versioned/scheme"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehelper"
+)
+
+const (
+	credentialsRequestGroup    = "cloudcredential.openshift.io"
+	credentialsRequestVersion  = "v1"
+	credentialsRequestResource = "credentialsrequests"
+
+	specHashAnnotation = "operator.openshift.io/spec-hash"
```

Comment:
please expose the const in resourceapply and use that value.

Reply:
Moved this code to resourceapply, so I don't need this const here anymore